data_steward/cdr_cleaner/cleaning_rules/remove_invalid_procedure_source_records.py

```diff
@@ -57,11 +70,25 @@ def get_remove_invalid_procedure_source_queries(project_id, dataset_id):
     """
     queries_list = []
 
-    query = dict()
-    query[cdr_consts.
-          QUERY] = REMOVE_INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
-              dataset=dataset_id, project=project_id)
-    queries_list.append(query)
+    # queries to sandbox
+    invalid_records = dict()
+    invalid_records[
+        cdr_consts.QUERY] = INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
+            project=project_id,
+            dataset=dataset_id,
+            sandbox_dataset=get_sandbox_dataset_id(dataset_id),
+            intermediary_table=INTERMEDIARY_TABLE_NAME)
+    queries_list.append(invalid_records)
+
+    # queries to delete invalid procedure source records
+    valid_records = dict()
+    valid_records[
+        cdr_consts.QUERY] = VALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY.format(
+            project=project_id,
+            dataset=dataset_id,
+            sandbox_dataset=get_sandbox_dataset_id(dataset_id),
+            intermediary_table=INTERMEDIARY_TABLE_NAME)
+    queries_list.append(valid_records)
```

Comment:
Assuming we are using a `SELECT` query to pick the records we want and overwriting the current table, we need to set the dictionary parameters for the destination dataset, destination table, and write disposition.  You can view the ` get_src_concept_id_logging_queries` function in `curation/data_steward/cdr_cleaner/cleaning_rules/replace_standard_id_in_domain_tables.py` for an example.  Otherwise, the query is just selecting and saving to a temporary table.

Reply:
Sorry this is a mess. You're right, this should be `DELETE FROM` but I had `SELECT` for testing purposes and never reverted back in the query. 

So in that case it should then be switched back to `IN`, see below, let me know if that makes sense.

```
DELETE FROM
`{project}.{dataset}.procedure_occurrence`
WHERE
procedure_occurrence_id IN (SELECT procedure_occurrence_id FROM `{project}.{sandbox_dataset}.{intermediary_table}`)
```

Reply:
Ok.  BigQuery works faster using Select statements.  So If you want to keep the select statement syntax (which I would suggest) you would need to set the extra query parameters to ovewrite the existing table.  Otherwise you can use a Delete statement without the additional query parameters and with the `IN` condition.

Reply:
I've gone ahead and updated the `SELECT` statement with the additional query parameters.